dto/src/main/java/ca/on/oicr/gsi/runscanner/dto/dragen/DragenAnalysis.java

```diff
@@ -0,0 +1,28 @@
+package ca.on.oicr.gsi.runscanner.dto.dragen;
+
+import ca.on.oicr.gsi.runscanner.dto.Analysis;
+import ca.on.oicr.gsi.runscanner.dto.dragen.samplesheet.Samplesheet;
+import java.util.HashMap;
+import java.util.Map;
+
+// Represents an entire attempt at a DRAGEN Pipeline
+public class DragenAnalysis implements Analysis {
+  Samplesheet samplesheet;
+  Map<String, DragenWorkflowAnalysis> analyses = new HashMap<>();
```

Comment:
This is another field that could be defined in the Analysis interface. I imagine all workflow analysis should have some common attributes - start and completion time? There could be a `WorkflowAnalysis` abstract class that defines these, and `Analysis` could take a type parameter to specify the implementation. e.g.

```
public abstract class Analysis<T extends WorkflowAnalysis>

public class DragenAnalysis extends Analysis<DragenWorkflowAnalysis>
```

I think the workflow name also belongs in the WorkflowAnalysis class. Standardizing as many fields as possible will make for a cleaner API that's easier to understand and work with.

Reply:
Done